Improve KMS documentation (#493)
digital-asset/docs.daml.com: docs/2.8.0/docs/canton/usermanual/kms/kms_gcp_setup.rst

- ``type`` specifies which KMS to use.
- ``location-id`` specifies which region the GCP KMS is bound to.
- ``project-id`` specifies which project are we binding to.
- ``key-ring-id`` specifies the keyring to use. Multi region keys are enabled for an entire keyring. Therefore, the KMS operator is responsible for setting the keyring correctly depending on the systems' needs.
- ``audit-logging`` flag that enables logging of every call made to the GCP KMS

Configure GCP Credentials and Permissions
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The authorized actions (i.e. IAM permissions) for the GCP KMS depend on the mode of operation we are running, either:
(a) :ref:`an encrypted private key storage using envelope encryption <encrypted_private_key_storage_permissions_gcp>`; or
(b) :ref:`externally storing and using keys directly from GCP KMS <external_key_storage_permissions_gcp>`.
A complete list is given in those sections.

For GCP, Canton uses a `GCP service account
<https://cloud.google.com/docs/authentication/provide-credentials-adc#local-user-cred>`_. For example,
the standard environment variable `GOOGLE_APPLICATION_CREDENTIALS` can be used after

> setting up a local Application Default Credentials (ADC) file for our service account.

Reviewer: ```suggestion
setting up a local Application Default Credentials (ADC) file for the service account.
```

The protection and rotation of
the credentials for GCP are the responsibility of the node operator.

Auditability
^^^^^^^^^^^^

> GCP provides tools to monitor KMS keys. For GCP you can refer to the `GCP official documentation
> <https://cloud.google.com/kms/docs/audit-logging>`_ for information on logging.

Reviewer: ```suggestion
GCP provides tools to monitor KMS keys. For GCP logging information you can refer to the `GCP official documentation
<https://cloud.google.com/kms/docs/audit-logging>`_.
```

> Errors resulting from the use of KMS keys are logged in Canton.

Reviewer: ```suggestion
Canton logs errors resulting from the use of KMS keys.
```

Logging
^^^^^^^

For further auditability, Canton can be configured to log every call made to the GCP KMS.
To enable this feature, set the ``audit-logging`` field of the KMS configuration to ``true``.

> By default, when using a file-based logging configuration, such logs will be written into the main canton log file.

Reviewer: ```suggestion
By default, when using a file-based logging configuration, such logs are written into the main Canton log file.
```
